### DISCUSSION TITLE
Add a maintenance mode to prevent logins

### DISCUSSION THREAD

**Discussion by luebbe** [WeblateOrg/weblate#7596] [Ideas]
We haven't found a built in function yet that allows us to tell users that an update is going to happen soon.

It would be nice to be able to set Weblate into a maintenance mode, which shows a big banner on the login screen and on all other screens telling the people not to log in and to save their work and log out. The maintenance mode could be turned off automatically when the weblate instance has restarted.

**Comment by nijel** [comment]
Something like https://docs.weblate.org/en/latest/admin/announcements.html?

**Comment by luebbe** [comment]
Almost. I see that I can set an expiry date there. Maybe I missed that when I tried it out a while ago.
When I tried it out, I didn't see the message. (Wrote a message in Browser A, opened Weblate in Browser B, didn't see it in either A or B).
Now I found out that the message is not visible while I'm in the "manage" area. Probably that's why I thought that it didn't work.

So in order to see the message in Weblate:
- you have to be logged in
- and you have to be outside the "manage" area

Personally I like the implementation of the Jenkins maintenance mode:
-  It displays a big red banner at the top of the screen, even if you are not logged in.
-  The banner/maintenance mode goes away automatically after a restart, so no need to set an expiry date/time or to remove the message manually after the update.

![grafik](https://user-images.githubusercontent.com/3872576/167079075-925d0213-d84b-4c93-9125-c5044891ce7c.png)
